src/Compilers/Core/Portable/PEWriter/PeWriter.cs

```diff
@@ -196,6 +199,15 @@ internal static class PeWriter
                 debugDirectoryBuilder = null;
             }
 
+            var strongNameProvider = context.Module.CommonCompilation.Options.StrongNameProvider;
```

Comment:
The `privateKeyOpt` is passed as a parameter but the `StrongNameProvider` is obtained from `Compilation.Options`. Is there a reason to handle the two differently?

Reply:
It seemed easier this way.  If the compilation is already getting passed down via context, why also pass it down separately?

Reply:
Then why pass in `privateKeyOpt`? Couldn't that be calculated here?
```
var privateKeyOpt =
    (strongNameProvider?.Capability == SigningCapability.SignsPeBuilder &&
        !compilation.Options.PublicSign) ?
    compilation.StrongNameKeys.PrivateKey :
    null;
```

Reply:
I wanted the logic for how to sign things to be mostly in one place.  Since stream signing is done in `Compilation`, I wanted at least the private key extraction to happen nearby.